consensus/dummy/dynamic_fees.go

```diff
@@ -143,34 +105,119 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header, timestamp uin
 		num.Div(num, parentGasTargetBig)
 		num.Div(num, baseFeeChangeDenominator)
 		baseFeeDelta := math.BigMax(num, common.Big1)
+
+		if timestamp < parent.Time {
+			// This should never happen as the fee window calculations should
+			// have already failed, but it is kept for clarity.
+			return nil, fmt.Errorf("cannot calculate base fee for timestamp %d prior to parent timestamp %d",
+				timestamp,
+				parent.Time,
+			)
+		}
+
 		// If timeElapsed is greater than [params.RollupWindow], apply the
 		// state transition to the base fee to account for the interval during
 		// which no blocks were produced.
 		//
 		// We use timeElapsed/params.RollupWindow, so that the transition is
 		// applied for every [params.RollupWindow] seconds that has elapsed
 		// between the parent and this block.
-		if timeElapsed > params.RollupWindow {
-			// Note: timeElapsed/params.RollupWindow must be at least 1 since
-			// we've checked that timeElapsed > params.RollupWindow
-			baseFeeDelta = new(big.Int).Mul(baseFeeDelta, new(big.Int).SetUint64(timeElapsed/params.RollupWindow))
+		var (
+			timeElapsed    = timestamp - parent.Time
+			windowsElapsed = timeElapsed / params.RollupWindow
+		)
+		if windowsElapsed > 1 {
+			bigWindowsElapsed := new(big.Int).SetUint64(windowsElapsed)
+			// Because baseFeeDelta could actually be [common.Big1], we must not
+			// modify the existing value of `baseFeeDelta` but instead allocate
+			// a new one.
+			baseFeeDelta = new(big.Int).Mul(baseFeeDelta, bigWindowsElapsed)
```

Comment:
This is technically a real change, but I think the code is much more clear (and it should result in the same behavior)

Reply:
I don't see any behavior differences.

Reply:
Technically the prior code would have multiplied by `1` where the new code just skips.

The prior code was:
```golang
if timeElapsed > params.RollupWindow {
```

The new code is essentially:
```golang
if timeElapsed >= 2*params.RollupWindow {
```